`assertTraitChanges` reports chained property events in reverse order

The following example demonstrates the problem. Please note that the calculation takes place correctly

```python
from traits.api import Int, HasStrictTraits, Property
from traits.testing.unittest_tools import unittest, UnittestTools


class MyClass(HasStrictTraits):

    value = Int
    
    new_value = Property(Int, depends_on='value')
    
    new_new_value = Property(Int, depends_on='new_value')
    
    def _get_new_value(self):
        print 'I was called first'
        return self.value + 10

    def _get_new_new_value(self):
        print 'I was called second' 
        return self.new_value + 10
        
    
class TestMyClass(unittest.TestCase, UnittestTools):

    
    def test_my_class(self):
    
        my_class = MyClass()
        
        traits = 'new_value,new_new_value'
        with self.assertTraitChanges(my_class, traits, count=2) as collected:
            my_class.value = 5
        names = [event[1] for event in collected.events]
        # this will fail with names having the reverse order.
        self.assertEqual(names, ['new_value', 'new_new_value'])


if __name__ == '__main__':
    unittest.main()
```

For the record, here's the output with current Traits master, under Python 2.7:

```
I was called first
I was called second
I was called first
F
======================================================================
FAIL: test_my_class (__main__.TestMyClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "traits_bug_212.py", line 34, in test_my_class
    self.assertEqual(names, ['new_value', 'new_new_value'])
AssertionError: Lists differ: ['new_new_value', 'new_value'] != ['new_value', 'new_new_value']

First differing element 0:
new_new_value
new_value

- ['new_new_value', 'new_value']
+ ['new_value', 'new_new_value']

----------------------------------------------------------------------
Ran 1 test in 0.005s

FAILED (failures=1)
```

--- a/traits/testing/unittest_tools.py
+++ b/traits/testing/unittest_tools.py
@@ -192,7 +192,7 @@ class UnittestTools(object):
 
     def assertTraitChanges(self, obj, trait, count=None, callableObj=None,
                            *args, **kwargs):
-        """Assert an object trait changes a given number of times.
+        """ Assert an object trait changes a given number of times.
 
         Assert that the class trait changes exactly `count` times during
         execution of the provided function.

--- a/traits/testing/unittest_tools.py
+++ b/traits/testing/unittest_tools.py
@@ -192,12 +192,12 @@ class UnittestTools(object):
 
     def assertTraitChanges(self, obj, trait, count=None, callableObj=None,
                            *args, **kwargs):
-        """ Assert an object trait changes a given number of times.
+        """Assert an object trait changes a given number of times.
 
         Assert that the class trait changes exactly `count` times during
         execution of the provided function.
 
-        Method can also be used in a with statement to assert that the
+        This method can also be used in a with statement to assert that
         a class trait has changed during the execution of the code inside
         the with statement (similar to the assertRaises method). Please note
         that in that case the context manager returns itself and the user
@@ -208,6 +208,11 @@ class UnittestTools(object):
 
         - All the fired events by accessing the ``events`` attribute of
           the return object.
+
+        Note that in the case of chained properties (trait 'foo' depends on
+        'bar', which in turn depends on 'baz'), the order in which the
+        corresponding trait events appear in the ``events`` attribute is
+        not well-defined, and may depend on dictionary ordering.
 
         **Example**::
 

